arch/arm/soc/atmel_sam0/Kconfig

```diff
@@ -13,6 +13,39 @@ config SOC_FAMILY
 	string
 	default atmel_sam0
 
+config SOC_ATMEL_SAMD_XOSC32K
+	bool "Enable the external 32 kHz crystal oscillator"
+	default n
+	help
+	  Say y to enable the external 32 kHZ crystal oscillator at
+	  startup.  This can then be selected as the main clock source
+	  for the SOC.
+
+config SOC_ATMEL_SAMD_XOSC
+	bool "Enable the external crystal oscillator"
+	default n
+	help
+	  Say y to enable the external crystal oscillator at startup.
+
+choice
+	prompt "Main clock source"
+	help
+	  Selects the clock that the main clocks, such as the CPU
+	  clock and AHB clock, will be derived from.
+
+config SOC_ATMEL_SAMD_XOSC32K_AS_MAIN
+	depends on SOC_ATMEL_SAMD_XOSC32K
+	bool "XOSC32K"
+
+config SOC_ATMEL_SAMD_XOSC_AS_MAIN
+	depends on SOC_ATMEL_SAMD_XOSC
+	bool "XOSC"
+
+config SOC_ATMEL_SAMD_OSC8M_AS_MAIN
```

Comment:
Settings in this file should be applicable to the whole SAM0 family, not only SAM Dxx series. SAM0 family has following members: B11, C20, C21, D09, D10, D11, D20, D21, DA1, HA1, L21, L22, R21, R30. Some of them don't have 8 MHz RC oscillator.

Reply:
Could we call it SAMD2X then?

Reply:
If we are able to reduce the number of configuration files by recognizing similarities between different chips that's probably good. We would however need to ensure that our implementation is correct and there are no (not yet implemented) series that will break it. Looking at the datasheets of D09, D10, D11, D20, D21, DA1 series they all seem to have identical clock subsystem. So using SAMD common clock options may be an option. But we would need to move it to some other Kconfig file as this one is meant for the whole SAM0 family. Also we would need to ensure that if we ever add e.g. B11 series the SAMD options will not be sourced.

That said this is not how other arch/arm/soc/* families/series are configured. So far every series has its own independent configuration files. It would be best to ask @galak or @MaureenHelm what's the best way to proceed.

Reply:
I've cleaned this up, as we see new SoCs added and new patterns we can tweak things.  for now I added a Kconfig.samd2x for that stuff.